vsintegration/src/FSharp.Editor/QuickInfoProvider.fs

```diff
@@ -0,0 +1,124 @@
+        override this.Create() : FrameworkElement =
+            let label = new Label()
+            label.Content <- content
+            label.Foreground <- SolidColorBrush(Colors.Black)
+            upcast label
+
+[<Shared>]
+[<ExportQuickInfoProvider(PredefinedQuickInfoProviderNames.Semantic, FSharpCommonConstants.FSharpLanguageName)>]
+type internal FSharpQuickInfoProvider [<System.ComponentModel.Composition.ImportingConstructor>] 
+    ([<System.ComponentModel.Composition.Import(typeof<SVsServiceProvider>)>] serviceProvider: IServiceProvider) =
+
+    let xmlMemberIndexService = serviceProvider.GetService(typeof<SVsXMLMemberIndexService>) :?> IVsXMLMemberIndexService
+    let documentationBuilder = XmlDocumentation.CreateDocumentationBuilder(xmlMemberIndexService, serviceProvider.DTE)
+
+    static member ProvideQuickInfo
+        (
+            document: Document, 
+            sourceText: SourceText, 
+            position: int, 
+            options: FSharpProjectOptions, 
+            filePath: string, 
+            textVersionHash: int,
+            cancellationToken: CancellationToken
+        ) = 
+        async {
+            let! parseResults = FSharpChecker.Instance.ParseFileInProject(filePath, sourceText.ToString(), options)
+            let! checkFileAnswer = FSharpChecker.Instance.CheckFileInProject(parseResults, filePath, textVersionHash, sourceText.ToString(), options)
+            let checkFileResults = 
+                match checkFileAnswer with
+                | FSharpCheckFileAnswer.Aborted -> failwith "Compilation isn't complete yet"
+                | FSharpCheckFileAnswer.Succeeded(results) -> results
+          
+            let textLine = sourceText.Lines.GetLineFromPosition(position)
+            let textLineNumber = textLine.LineNumber + 1 // Roslyn line numbers are zero-based
+            let textLinePos = sourceText.Lines.GetLinePosition(position)
+            let textLineColumn = textLinePos.Character
+            //let qualifyingNames, partialName = QuickParse.GetPartialLongNameEx(textLine.ToString(), textLineColumn - 1)
+            let defines = CompilerEnvironment.GetCompilationDefinesForEditing(document.Name, options.OtherOptions |> Seq.toList)
+            let tryClassifyAtPosition position = 
+                CommonHelpers.tryClassifyAtPosition(document.Id, sourceText, filePath, defines, position, cancellationToken)
+            
+            let quickParseInfo = 
+                match tryClassifyAtPosition position with 
+                | None when textLineColumn > 0 -> tryClassifyAtPosition (position - 1) 
+                | res -> res
+
+            match quickParseInfo with 
+            | Some (islandColumn, qualifiers) -> 
+                //let tokenTag = QuickParse.CorrectIdentifierToken "" tokenTag
+                let! res = checkFileResults.GetToolTipTextAlternate(
+                               textLineNumber, islandColumn, textLine.ToString(), qualifiers, 189)
```

Comment:
What is 189?  The tokenTag I guess. I think you can use ``tagOfToken`` on an identifier